kotlin/goodmetrics/src/main/kotlin/goodmetrics/MetricsFactory.kt

```diff
@@ -60,12 +60,14 @@ class MetricsFactory(
     /**
      * Passes a Metrics into your scope. Record your unit of work; when the scope exits
      * the Metrics will be stamped with `totaltime` and emitted through the pipeline.
+     *
+     * If you don't want `totaltime` timeseries data, then specify `metricBehavior: MetricBehavior.NO_TOTALTIME`.
      */
-    inline fun <T> record(name: String, stampAt: TimestampAt = TimestampAt.Start, block: (Metrics) -> T): T {
+    inline fun <T> record(name: String, stampAt: TimestampAt = TimestampAt.Start, metricsBehavior: MetricsBehavior = MetricsBehavior.DEFAULT, block: (Metrics) -> T): T {
```

Comment:
Note that this change will break consumers in your classpath if your consumers do not align jar versions. (I happen to know yours will align because I made them align _but_ food for thought)

Reply:
Updated after discussion in [91c16e](https://github.com/kvc0/goodmetrics_kotlin/pull/30/commits/891c16e7d1898b75d687b0c76ce08aaee3bc2ea7)